Lib/test/test_unicode.py

```diff
@@ -185,6 +185,18 @@ def test_iterators(self):
         self.assertEqual(next(it), "\u3333")
         self.assertRaises(StopIteration, next, it)
 
+    def test_iterators_invocation(self):
+        cases = [type(iter('abc')), type(iter('🚀'))]
+        for cls in cases:
+            with self.subTest(cls=cls):
+                self.assertRaises(TypeError, cls)
+
+    def test_iteration(self):
+        cases = ['abc', '🚀🚀🚀', "\u1111\u2222\u3333"]
+        for case in cases:
+            with self.subTest(string=case):
+                self.assertEqual(case, "".join(iter(case)))
+
```

Comment:
Maybe add a test that already-exhausted iterators continue raising StopIteration over and over; test the `if (seq == NULL)` branch.

Reply:
Also add a test for pickling iterator objects to exercise the `unicodeiter_reduce` and `unicodeiter_setstate` functions.

Reply:
done